metrics-core/src/main/java/com/yammer/metrics/stats/ExponentiallyDecayingSample.java

```diff
@@ -128,14 +135,24 @@ private void rescaleIfNeeded() {
     public Snapshot getSnapshot() {
         lockForRegularUsage();
         try {
-            return new Snapshot(values.values());
+            Collection<ValueWithTimestamp> valuesWithTimestamps = values.values();
+            int now = currentTimeInSeconds();
+
+            List<Long> values = new ArrayList<Long>();
+            for (ValueWithTimestamp valueWithTimestamp : valuesWithTimestamps) {
+                int age = now - valueWithTimestamp.getTimestamp();
+                if (age < maxAge) {
+                    values.add(valueWithTimestamp.getValue());
+                }
+            }
+            return new Snapshot(values);
         } finally {
             unlockForRegularUsage();
         }
     }
 
-    private long currentTimeInSeconds() {
-        return TimeUnit.MILLISECONDS.toSeconds(clock.time());
+    private int currentTimeInSeconds() {
+        return (int) TimeUnit.MILLISECONDS.toSeconds(clock.time());
```

Comment:
This cast makes me slightly nervous – any chance of using guava's `Ints.checkedCast()` or a copypasta of that method? ...or at least do _something_ if this method returns a negative value?

https://en.wikipedia.org/wiki/Year_2038_problem